metadata-ingestion/src/datahub/ingestion/source/dremio/dremio_config.py

```diff
@@ -0,0 +1,197 @@
+import os
+from typing import List, Literal, Optional
+
+import certifi
+from pydantic import Field, validator
+
+from datahub.configuration.common import AllowDenyPattern, ConfigModel
+from datahub.ingestion.source.ge_profiling_config import GEProfilingConfig
+from datahub.ingestion.source.state.stateful_ingestion_base import (
+    StatefulIngestionConfigBase,
+)
+from datahub.metadata.schema_classes import FabricTypeClass
+
+
+class ProfileConfig(GEProfilingConfig):
+    partition_profiling_enabled: bool = Field(
+        default=False,
+        description="Partition profiling disabled for Dremio.",
+    )
+    include_field_median_value: bool = Field(
+        default=False,
+        description="Median causes a number of issues in Dremio.",
+    )
+    query_timeout: int = Field(
+        default=300, description="Time before cancelling Dremio profiling query"
+    )
+
+    row_count: bool = True
+    column_count: bool = True
+    sample_values: bool = True
+
+
+class DremioSourceMapping(ConfigModel):
+    platform: Optional[str] = Field(
+        default=None,
+        description="Source connection made by Dremio (e.g. S3, Snowflake)",
+    )
+    platform_name: Optional[str] = Field(
+        default=None,
+        description="Alias of platform in Dremio connection",
+    )
+    platform_instance: Optional[str] = Field(
+        default=None,
+        description="Platform instance of source connection in Datahub",
+    )
+    dremio_source_type: Optional[str] = Field(
+        default=None,
+        description="Source connection made by Dremio (e.g. S3, Snowflake)",
+    )
+    env: Optional[str] = Field(
+        default=FabricTypeClass.PROD,
+        description="ENV in Datahub of source connection made by Dremio (e.g. PROD)",
+    )
+    root_path: Optional[str] = Field(
+        default=None,
+        description="Root path of source - Extracted from Dremio API",
+        hidden_from_docs=True,
+    )
+    database_name: Optional[str] = Field(
+        default=None,
+        description="Database of source - Extracted from Dremio API",
+        hidden_from_docs=True,
+    )
+
```

Comment:
Can we make `platform`, `dremio_source_type`, `env` optional ?

What is difference between platform and platform_name here ?

Reply:
platform here could be s3/glue in short its sources that we added into dremio
platform_name is name of the source.

platform, dremio_source_type, env all this are optional now in config.

Removed env as it wont required.